server/app/routes/api/v1/va/ambassadors.js

```diff
@@ -201,12 +201,18 @@ async function approveAmbassador(req, res) {
  *
  */
 async function updateW9Ambassdor(req, res) {
+  //first, search for the Ambassador
   let found = await req.neode.first('Ambassador', 'id', req.params.ambassadorId)
+  let w9found = await req.neode.first('Ambassador', 'has_w9', req.params.has_w9)
```

Comment:
I think we're probably going to want to make this whole thing a single cypher query instead of using neode, actually.

So what is happening right now is that line 205 finds an Ambassador with the ID you give. Then line 206 finds the first Ambassador in the db with `has_w9` equal to the `req.params.has_w9` request param.

Then on 212, this is saying if we didn't find a matching Ambassador in the db on line 206, then this raw cypher query on line 213 is finding the same Ambassador from line 205 again (but in raw cypher this time, not a neode object), and setting its `has_w9` attribute to the appropriate request param.

Then on 217, we're basically doing it again, but with the first neode object from line 205.

So actually, I think you can basically delete line 205 and 206, and take line 213 out of the conditional, and delete line 216 and 217. I think that will do what we want in just that one raw cypher query that you have here. Maybe check with Lee to confirm.

Reply:
yea it's overkill but i put both ways to see which was preferred. Agree that just the cypher is the better way